nuimo/src/main/kotlin/com/senic/nuimo/NuimoBluetoothController.kt

```diff
@@ -117,9 +123,23 @@ class NuimoBluetoothController(bluetoothDevice: BluetoothDevice, context: Contex
             }
         }
 
+        override fun onCharacteristicRead(gatt: BluetoothGatt, characteristic: BluetoothGattCharacteristic, status: Int) {
+            if (characteristic.uuid.equals(BATTERY_CHARACTERISTIC_UUID)) {
+                batteryPercentage = characteristic.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 0) ?: -1
+                notifyListeners { it.onBatteryPercentageChanged(batteryPercentage) }
+                matrixWriter?.onWrite()
```

Comment:
`matrixWriter?.onWrite` is the first hing that should happen to immediately allow for the next write request. See also `onCharacteristicWrite`.